plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/AuthorizationCodeReceiver.java

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.google.cloud.tools.eclipse.appengine.login;
+
+import com.google.common.base.Preconditions;
+
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Runs a Jetty web server instance that opens up an HTTP listening port to receive OAuth2
+ * authorization code.
+ */
```

Comment:
I'm not actually familiar with the google api client library, but I do
remember that they repackage a few things, I guess as part of providing an
api and support classes for what we're doing here.

@loosebazooka may know more.

On Tue, Aug 9, 2016 at 11:30 AM, Chanseok Oh <notifications@github.com>
wrote:

> In plugins/com.google.cloud.tools.eclipse.appengine.login/
> src/com/google/cloud/tools/eclipse/appengine/login/
> AuthorizationCodeReceiver.java
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/513#discussion_r74082031>
> :
>
> > +import org.eclipse.jetty.server.Request;
> > +import org.eclipse.jetty.server.Server;
> > +import org.eclipse.jetty.server.ServerConnector;
> > +import org.eclipse.jetty.server.handler.AbstractHandler;
> > +
> > +import java.io.IOException;
> > +import java.util.logging.Level;
> > +import java.util.logging.Logger;
> > +
> > +import javax.servlet.http.HttpServletRequest;
> > +import javax.servlet.http.HttpServletResponse;
> > +
> > +/**
> > + * Runs a Jetty web server instance that opens up an HTTP listening port to receive OAuth2
> > + * authorization code.
> > + */
>
> Ah, I see there is this VerificationCodeReceiver. I'll add that.
>
> For Server, I believe it's really org.eclipse.jetty.server.Server (or
> org.mortbay.jetty.Server in the old code), isn't it? Their APIs are
> identical. I think the reason it has the package name
> com.google.api.client.repackaged.org.mortbay.jetty.Server is that someone
> repackages this Jetty code and other login-related jars into a single jar
> called google-api-java-client-min-repacakged-1.20.0.jar, right?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools/pull/513/files/60793f3ef568f6fd00a5954c61e68787dd053ce6#r74082031>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HSiAO4STe6Bs9kuiaNrB7xVnGzAcks5qeJ0igaJpZM4JeiAx>
> .
>